etc/kayobe/ansible/deploy-baremetal-instance.yml

```diff
@@ -0,0 +1,99 @@
+---
+
+- name: Update network allocations for new hypervisors
+  hosts: compute
+  gather_facts: false
+  connection: local
+  serial: 1
+  vars:
+    network_allocation_path: "{{ kayobe_env_config_path }}/network-allocation.yml"
+  tasks:
+    - name: Show baremetal node
+      ansible.builtin.shell:
+        cmd: "openstack baremetal node show {{ inventory_hostname }} -f json"
+      register: node_show
+      failed_when: false
+      changed_when: false
+      delegate_to: localhost
+
+    - name: Set baremetal node JSON variable
+      ansible.builtin.set_fact:
+        node_show_json: "{{ node_show.stdout | to_json | from_json }}"
+      failed_when: false
+      changed_when: false
+
+    - block:
+        - name: Slurp network allocations
+          ansible.builtin.slurp:
+            path: "{{ network_allocation_path }}"
+          register: net_alc
+
+        - name: Read network allocations
+          ansible.builtin.set_fact:
+            net_alc_yaml: "{{ net_alc['content'] | b64decode | from_yaml }}"
+
+        - name: Write node IP address to allocations
+          ansible.builtin.set_fact:
+            new_net_alc: "{{ net_alc_yaml | combine(new_ips, recursive=True) }}"
+          vars:
+            new_ips: "{ '{{ admin_oc_net_name }}_ips': { '{{ inventory_hostname }}': '{{ ansible_host }}' } }"
```

Comment:
The net IPs are dictionary vars you have access to right? I don't think you need to slurp them in from a file